tap_typeform/streams.py

```diff
@@ -292,76 +318,52 @@ def sync_forms(atx):
-            LOGGER.info('ut_current_date: {} '.format(ut_current_date))
-            ut_next_date = int(next_date.timestamp())
-            LOGGER.info('ut_next_date: {} '.format(ut_next_date))
-            [responses, max_submitted_at] = sync_form(atx, form_id, ut_current_date, ut_next_date)
-            # if the max responses were returned, we have to make the call again
-            # going to increment the max_submitted_at by 1 second so we don't get dupes,
-            # but this also might cause some minor loss of data.
-            # there's no ideal scenario here since the API has no other way than using
-            # time ranges to step through data.
-            bookmark_value = construct_bookmark(stream_name, next_date)
-
-            while responses == 1000:
-                interim_next_date = pendulum.parse(max_submitted_at) + datetime.timedelta(seconds=1)
-                ut_interim_next_date = int(interim_next_date.timestamp())
-                bookmark_value = construct_bookmark(stream_name, interim_next_date)
-                write_forms_state(atx, stream_name, form_id, bookmark_value)
-                [responses, max_submitted_at] = sync_form(atx, form_id, ut_interim_next_date, ut_next_date)
-
-            # check if next_date is in the future
-            now_parsed = pendulum.parse(now.isoformat())
-            next_date = now_parsed if next_date > now else next_date
-
-            # if the prior sync is successful it will write the date_to_resume bookmark
+        page_count = 0
+
+        ut_current_date = int(current_date.timestamp())
+        LOGGER.info('ut_current_date: {} '.format(ut_current_date))
+        
+        page_count, max_submitted_at, max_token = sync_form(atx, form_id, ut_current_date)
+
+        parsed_max_submitted_at = pendulum.parse(max_submitted_at)
+        bookmark_value = construct_bookmark(stream_name, parsed_max_submitted_at)
+
+        # if the max responses were returned, we have to make the call again
+        # using the max_token from the initial call.
+        while page_count > 1:
+            page_count, max_submitted_at, max_token = sync_form(atx,
+                                                                form_id,
+                                                                ut_current_date,
+                                                                token=max_token,
+                                                                next_page=True)
+
+            parsed_max_submitted_at = pendulum.parse(max_submitted_at)
+            bookmark_value = construct_bookmark(stream_name, parsed_max_submitted_at)
             write_forms_state(atx, stream_name, form_id, bookmark_value)
-            current_date = next_date
 
-    if 'forms'in atx.selected_stream_ids:
+        # check if bookmark is in the past
+        now_parsed = pendulum.parse(now.isoformat())
+        updated_bookmark_value = now_parsed if parsed_max_submitted_at < now_parsed else parsed_max_submitted_at
```

Comment:
Do you think this would be cleaner as `max(parsed_max_submitted_at, now_parsed)`?

Reply:
agreed